{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TAG分析\n",
    "TAG用于分析单细胞混样数据， 可以根据tag barcode sequence拆分样本，需要先使用CeleScope分析RNA数据。\n",
    "\n",
    "- sample：生成样本信息\n",
    "- barcode：解复用barcode，主要过滤fastq文件中不包含barcode的reads和低质量的reads\n",
    "- cutadapt：对reads进行trimming\n",
    "- mapping_tag：根据tag barcode sequence对cell barcodes进行分配\n",
    "- count_tag：统计分配到不同tag barcode sequence的细胞数\n",
    "- analysis_tag：将scRNA聚类信息与标签分配信息相结合的可视化\n",
    "- split_tag：根据tag barcode sequence拆分fastq文件，表达矩阵，VDJ结果文件等\n",
    "\n",
    "1.对样本进行celescope转录组分析\n",
    "\n",
    "2.创建mapfile，barcode_fasta文件，multi_tag命令生成shell文件\n",
    "\n",
    "multi_tag \\\n",
    "\n",
    "    --mapfile tag.mapfile\\\n",
    "\n",
    "    --barcode_fasta tag_barcode.fasta\\\n",
    "\n",
    "    --fq_pattern L25C15\\\n",
    "\n",
    "    --split_matrix\\\n",
    "\n",
    "    --mod shell\n",
    "\n",
    "- --split_matrix代表拆分表达矩阵，还支持--split_fastq， --split_fl_vdj， --split_vdj分别拆分fastq文件，vdj分析和全长vdj分析结果。\n",
    "\n",
    "- cat tag_barcode.fasta\n",
    "\n",
    "&gt;TAG_1\n",
    "\n",
    "CGTGTTAGGGCCGAT\n",
    "\n",
    "&gt;TAG_2\n",
    "\n",
    "GAGTGGTTGCGCCAT\n",
    "\n",
    "&gt;TAG_3\n",
    "\n",
    "AAGTTGCCAAGGGCC\n",
    "\n",
    "&gt;TAG_4\n",
    "\n",
    "TAAGAGCCCGGCAAG\n",
    "\n",
    "&gt;TAG_5\n",
    "\n",
    "TGACCTGCTTCACGC\n",
    "\n",
    "&gt;TAG_6\n",
    "\n",
    "GAGACCCGTGGAATC\n",
    "\n",
    "- cat tag.mapfile\n",
    "\n",
    "tag&emsp;../../celescope_test_data/tag/fastqs/&emsp;test1&emsp;../../celescope_test_data/tag/fake_match_dir/\n",
    "\n",
    "3.运行生成的shell文件\n",
    "\n",
    "sh shell/test1.sh\n",
    "\n",
    "\n",
    "celescope tag sample --outdir .//test1/00.sample --sample test1 --thread 4 --chemistry auto  --fq1 ../../celescope_test_data/tag/fastqs/tag_1.fq.gz\n",
    "\n",
    "celescope tag barcode --outdir .//test1/01.barcode --sample test1 --thread 4 --chemistry auto --lowNum 2  --fq1 ../../celescope_test_data/tag/fastqs/tag_1.fq.gz --fq2 ../../celescope_test_data/tag/fastqs/tag_2.fq.gz\n",
    "\n",
    "celescope tag cutadapt --outdir .//test1/02.cutadapt --sample test1 --thread 4 --minimum_length 20 --nextseq_trim 20 --overlap 10 --insert 150  --fq .//test1/01.barcode/test1_2.fq\n",
    "\n",
    "celescope tag mapping_tag --outdir .//test1/03.mapping_tag --sample test1 --thread 4 --fq_pattern L25C45 --barcode_fasta ./smk_barcode.fa  --fq .//test1/02.cutadapt/test1_clean_2.fq\n",
    "\n",
    "celescope tag count_tag --outdir .//test1/04.count_tag --sample test1 --thread 4 --UMI_min auto --dim 1 --SNR_min auto --coefficient 0.1  --match_dir ../../celescope_test_data/tag/fake_match_dir/ --read_count_file .//test1/03.mapping_tag/test1_read_count.tsv\n",
    "\n",
    "celescope tag analysis_tag --outdir .//test1/05.analysis_tag --sample test1 --thread 4  --match_dir ../../celescope_test_data/tag/fake_match_dir/ --tsne_tag_file .//test1/04.count_tag/test1_tsne_tag.tsv\n",
    "\n",
    "celescope tag split_tag --outdir .//test1/06.split_tag --sample test1 --thread 4 --split_matrix  --match_dir ../../celescope_test_data/tag/fake_match_dir/ --umi_tag_file .//test1/04.count_tag/test1_umi_tag.tsv\n",
    "```\n",
    "\n",
    "# reference\n",
    "\n",
    "- test_script:\n",
    "\n",
    "https://github.com/singleron-RD/celescope_test_script\n",
    "\n",
    "- test_data:\n",
    "\n",
    "https://github.com/singleron-RD/celescope_test_data"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.6 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "name": "python",
   "version": "3.10.6"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "beedbe2faf2f7048d727558d0bc3221e7eba2a0b921cac4d4771b2feb8f74b30"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
